LICENSES/README

```diff
@@ -0,0 +1,60 @@
+The systemd project uses a single line reference to Unique License Identifiers
+in source files as defined by the Linux Foundation's SPDX project
+(https://spdx.org/).
+
+The current set of valid, predefined SPDX identifiers can be found on the SPDX
+License List at https://spdx.org/licenses/.
+
+The 'license/' directory contains all the licenses used by the sources included
+in the systemd project source tree.
+
+Unless otherwise noted, the systemd project sources are licensed under the terms
+and conditions of the "GNU Lesser General Public License v2.1 or later".
+
+The following exceptions apply:
+
+ * the udev sources are licensed under GPL-2.0-or-later
+ * the header files contained in src/basic/linux/ and src/shared/linux/ are
+   copied verbatim from the Linux kernel source tree and are licensed under
+   GPL-2.0 WITH Linux-syscall-note and are used within the scope of the
+   Linux-syscall-note exception, and thus their license does not affect the rest
+   of the source code/binaries
+ * the src/shared/initreq.h header is licensed under original license,
+   LGPL-2.0-or-later.
+ * the src/shared/linux/bpf_insn.h header is copied from the Linux kernel
+   source tree and is licensed under either BSD-2-Clause or GPL-2.0-only,
+   and thus is included in the systemd build under the BSD-2-Clause license.
+ * The src/basic/linux/wireguard.h header is copied from the Linux kernel
+   source tree and is licensed under either MIT or GPL-2.0 WITH
+   Linux-syscall-note, and thus is included in the systemd build under the MIT
+   license.
+ * the following sources are licensed under the MIT license:
+   - hwdb.d/parse_hwdb.py
+   - man/glib-event-glue.c
+   - src/basic/linux/batman_adv.h
+   - src/basic/sparse-endian.h
+   - tools/catalog-report.py
+ * the following sources are licensed under the CC0-1.0 license:
+   - src/basic/siphash24.c
+   - src/basic/siphash24.h
+   - src/systemctl/systemd-sysv-install.SKELETON
+   - tools/check-includes.pl
+ * the following sources are licensed under the LicenseRef-murmurhash2-public-domain:
+   - src/basic/MurmurHash2.c
+   - src/basic/MurmurHash2.h
+ * the following sources are licensed under the LicenseRef-lookup3-public-domain:
+   - src/libsystemd/sd-journal/lookup3.c
+   - src/libsystemd/sd-journal/lookup3.h
+
+OpenSSL notes
+
+Note that when building the systemd project with OpenSSL 3.0, the resulting
+binaries that link with libssl/libcrypto have to effectively be distributed
+under LGPL-3.0-or-later to be compatible with OpenSSL's Apache2 license.
+This does not affect the libsystemd.so shared library, which will not link
+with OpenSSL 3.0 and thus can still be distributed under LGPL-2.1-or-later
+in all cases, so that external programs linking to it are guaranteed to be
+unaffected.
```

Comment:
hmm, is this actually really the case? For me:

```
$ ldd ./build/libsystemd.so
	linux-vdso.so.1 (0x00007fff6cbe7000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f481abe3000)
	liblzma.so.5 => /lib64/liblzma.so.5 (0x00007f481abb7000)
	libzstd.so.1 => /lib64/libzstd.so.1 (0x00007f481aac1000)
	liblz4.so.1 => /lib64/liblz4.so.1 (0x00007f481aa9d000)
	libcap.so.2 => /lib64/libcap.so.2 (0x00007f481aa93000)
	libgcrypt.so.20 => /lib64/libgcrypt.so.20 (0x00007f481a957000)
	libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007f481a667000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f481a646000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f481a477000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f481adee000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f481a470000)
	libgpg-error.so.0 => /lib64/libgpg-error.so.0 (0x00007f481a44a000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f481a430000)
```
i.e. libcrypto is in there.

Reply:
uh, that's not good...

Reply:
it's src/libsystemd/sd-id128/sd-id128.c for HMAC - switch to khash like the rest of libsystemd?

Reply:
there's already an `#ifdef openssl ... #else` that uses khash

Reply:
I'll take this note out and do a separate PR